src/parser/vast_parser.js

```diff
@@ -231,20 +231,9 @@ export class VASTParser extends EventEmitter {
    * Returns a Promise which resolves with the array or rejects with an error according to the result of the parsing.
    * @param  {Object} vastXml - An object representing an xml document.
    * @param  {Object} options - An optional Object of parameters to be used in the parsing process.
-   * @emits  VASTParser#VAST-resolving
-   * @emits  VASTParser#VAST-resolved
    * @return {Promise}
    */
-  parse(
-    vastXml,
-    {
-      resolveAll = true,
-      wrapperSequence = null,
-      originalUrl = null,
-      wrapperDepth = 0,
-      isRootVAST = false
-    }
-  ) {
+  parseXML(vastXml, { isRootVAST = false }) {
```

Comment:
Since we removed any async operation we can make this method simply return an array of ads or throwing an Error in case the parsing fails

Reply:
And I would maybe rename this function to something more explicit like `parseVastXml`

Reply:
@eliamaino-fp 
So, how should I use it in `parse()`? 🤔
Personally I like to have consistent returns of promises in a function so if we treat `parseVastXml` as not async we should probably try/catch it in  `parse()` and if we get an error return a rejected promise? Unless we assume parse() is called within a chain of course.

Reply:
I know that might sound not that important, but if we plan to provide `parseVastXml ` as part of the public API and the code is not asynchronous we should not return a Promise (since users may really need to have this code executing in a sync way).

Assuming that `parse` will be called only in a Promise chain may be risky too :/

I'd suggest to go for a `try/catch` inside `parse`, something like:
```javascript
parse(vastXml, /* Other params */) {
  let ads = []

  try {
    ads = this.parseXML(vastXml, { isRootVAST })
  } catch (e) {
    return Promise.reject(e)
  }

  // rest of the logic
}
```

What do you think @bwallberg?

Reply:
@eliamaino-fp 
Fair enough :)
I've made the changes.